src/RFC-0205_LedgerTransactions.md

```diff
@@ -0,0 +1,148 @@
+
+## Implementation
+
+### Entities
+Normal transactions have only a single entity, the wallet which controls all secrets and transactions. But with hardware wallets, we need to define two distinct types:
+* Signer: This is the entity that keeps the secrets for the transactions and approves them, aka the hardware wallet.
+* Helper: This entity is the program that helps the signer construct the transaction, send it over the network and scan the network, aka wallet.
+
+### Process Overview
+By splitting the ownership of the [UTXO]'s secret by assigning knowledge of only the script key (\\( k_s \\) ) to the signer, we can lift much of the heavy cryptography like bulletproof creation to the helper device by exposing (\\( k_i \\) ) to it. By looking at how [one-sided-stealth] transactions are created, we can construct the script key in such a way that the helper can calculate the public script key, but cannot calculate the private script key.
+
+All [UTXO]s created will be created with a script `PushPubkey(K_S)`. The key (\\( k_s \\) ) is created as follows:
+$$
+\begin{aligned}
+k_S = k_i + a \\\\
+K_S = k_i \cdot G + A
+\end{aligned}
+$$
+
+The blinding factor (\\( k_i \\) ) is used as a random nonce when creating the script key. This means the helper can create the public key without the signer present, and the signer can then at a later stage create the private key from the nonce. The key pair (\\( a, A \\) ) is the master key pair from the signer. The private key (\\( a \\) ) is kept secret by the signer at all times.
+
+### Initialization
+Adding a hardware wallet to a wallet we need to ensure that all keys are only derived from a single seed phrase provided by the hardware wallet. 
+
+Helper asks signer for master helper key (\\( k_H \\) ). 
+This key is derived from the signer seed phrase.
+
+### Transaction receiving
+When a transaction is received the helper constructs the new [UTXO] with its Rangeproof. Choosing a new (\\( k_i \\) ) for the [UTXO], it calculates a new \\( s \\) and \\( K_S \\). It attaches the script `PushPubkey(K_S)` to output.
+
+### Transaction sending
+When the user wants to send a transaction, the helper retrieves the desired [UTXO]. The helper gives asks the signer to sign the transaction. 
+The singer calculates \\( k_s \\) to sign the transaction. 
+The signer creates a random nonce \\( k_O \\) to use for the script_offset. It produces the metadata signature with \\( k_O \\), and supplies the script_offset to the helper. 
+The helper can attach the correct signatures to the [UTXO]s and ship the transaction.
+
+### Receiving normal one-sided transaction
+This can be done by the helper asking the signer for a public key. And advertising this public key as the destination public key for a 1-sided transaction. 
+The helper can scan the blockchain for this public key. 
+
+### Receiving one-sided-stealth
+`TODO`
+
+### Output recovery
```

Comment:
Perhaps worth noting that this process only works for interactive outputs.

For legacy one-sided outputs, the helper can watch for `A` in scripts (as you note elsewhere), but since it can't complete the DHKE required to decrypt the commitment opening, it can't assert the output is valid and must send it to the signer to be checked.

For stealth one-sided outputs, the helper can't do anything except send such outputs to the signer to be checked.

Reply:
Yeah, for legacy one-sided it's mostly a UX issue, and somewhat a "wait for the signer" issue. For stealth one-sided it's an entirely different problem that does not scale.